src/main/java/no/uio/ifi/localega/doa/services/AAIService.java

```diff
@@ -33,25 +37,37 @@
     private String visaPublicKeyPath;
 
     /**
-     * Extracts datasets access information from the JWT token.
+     * Extracts datasets access information from the JWT or opaque access token.
      *
-     * @param jwtToken JWT token with access info in GA4GH Passport format.
+     * @param accessToken JWT or opaque access token.
      * @return IDs of datasets user has access to.
      */
-    public Collection<String> getDatasetIds(String jwtToken) {
-        DecodedJWT decodedJWT = JWT.decode(jwtToken);
-        boolean isVisa = decodedJWT.getClaims().containsKey("ga4gh_visa_v1");
+    public Collection<String> getDatasetIds(String accessToken) {
         Collection<Visa> visas = new ArrayList<>();
-        if (isVisa) {
-            getVisa(jwtToken).ifPresent(visas::add);
-        } else {
-            visas.addAll(getVisas(jwtToken));
+        if (StringUtils.countMatches(accessToken, '.') == 3) { // JWT access token
```

Comment:
```suggestion
        if (StringUtils.countMatches(accessToken, '.') == 2) { // JWT access token
```
JWT has 3 parts separated by 2 dots ... as pointed by @teemukataja